### PR TITLE
Using secret provider

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

**Review by ambiknai — COMMENTED** [review]
lgtm

**Review by sameshai — DISMISSED** [review]

File: pkg/nodeupdater/utils.go [hunk 1]
```diff
@@ -55,43 +51,31 @@ const (
...
 	storageSecretConfig := &StorageSecretConfig{
-		APIKey:              conf.VPC.G2APIKey,
-		IamTokenExchangeURL: fmt.Sprintf("%s/oidc/token", conf.VPC.G2TokenExchangeURL),
-		RiaasEndpointURL:    riaasInstanceURL,
-		BasicAuthString:     fmt.Sprintf("%s:%s", conf.VPC.IamClientID, conf.VPC.IamClientSecret),
+		RiaasEndpointURL: riaasInstanceURL,
 	}
 
-	accessToken, err := storageSecretConfig.GetAccessToken(ctxLogger)
+	// TODO - send appropriate reason for call
```
Comment (sameshai) [inline pkg/nodeupdater/utils.go]: We can tag it as used for VPCNodeLabelUpdator

Comment (GunaKKIBM) [reply pkg/nodeupdater/utils.go]: Added reason in GetDefaultIAMToken - "vpc-node-label-updater".

**Review by prankulmahajan — APPROVED** [review]
lgtm

**Pull request merged**